synapse/storage/state.py

```diff
@@ -725,7 +810,7 @@ def _get_state_for_groups(self, groups, types=None, filtered_types=None):
                 types_to_fetch = types
 
             group_to_state_dict = yield self._get_state_groups_from_groups(
-                missing_groups, types_to_fetch
+                missing_groups, types_to_fetch, cache == self._state_group_members_cache,
```

Comment:
this is a bit horrid and feels a bit backwards.

Can you pass a bool param (`use_members_cache`?) into this function instead of `cache`?

Reply:
not sure i follow. passing the right cache object into `_get_state_for_groups_using_cache` seems fine to me, and simplifies all the other cache references to just be `cache.get` etc? And `_get_state_groups_from_groups` is already taking the `bool|None` to say whether it should be limiting to members or non-members or not.  Or is the problem just the inlining the boolean expression?

Reply:
sorry, this was unclear.

I'm suggesting:

rather than pass a `cache` into `_get_state_for_groups_using_cache`, pass a bool which is True to use `_state_group_members_cache` and False to use `_state_group_cache`. Then on the first line (or wherever) of `_get_state_for_groups_using_cache`, do

```
cache = self._state_group_members_cache if use_members_cache else self._state_group_cache
```

and then on this line here you can just use `use_members_cache` instead of going back to a bool.

It just felt odd to be going back to a bool here. Though I don't feel that strongly about it so if you want to leave it alone that's fine.

Reply:
okay, that's how i interpreted, but am unsure that a cryptic bool flying around the place is better than saying "use this cache please", even though it does mean we end up with the slightly backwards comparison here back to a bool. i'd rather leave it as is if you're borderline.

Reply:
ok.